src/theory/theory_inference_manager.cpp

```diff
@@ -90,21 +92,52 @@ void TheoryInferenceManager::conflictExp(PfRule id,
 {
   if (!d_theoryState.isInConflict())
   {
-    if (d_pfee != nullptr)
-    {
-      // use proof equality engine to construct the trust node
-      TrustNode tconf = d_pfee->assertConflict(id, exp, args);
-      d_out.trustedConflict(tconf);
-    }
-    else
-    {
-      // version without proofs
-      Node conf = mkExplainPartial(exp, {});
-      conflict(conf);
-    }
+    // make the trust node
+    TrustNode tconf = mkConflictExp(id, exp, args);
+    // send it on the output channel
+    trustedConflict(tconf);
   }
 }
 
+TrustNode TheoryInferenceManager::mkConflictExp(PfRule id,
+                                                const std::vector<Node>& exp,
+                                                const std::vector<Node>& args)
+{
+  if (d_pfee != nullptr)
+  {
+    // use proof equality engine to construct the trust node
+    return d_pfee->assertConflict(id, exp, args);
+  }
+  // version without proofs
+  Node conf = mkExplainPartial(exp, {});
+  return TrustNode::mkTrustConflict(conf, nullptr);
+}
+
+void TheoryInferenceManager::conflictExp(const std::vector<Node>& exp,
+                                         ProofGenerator* pg)
+{
+  if (!d_theoryState.isInConflict())
```

Comment:
It feels wrong to me to silently drop conflicts, just because we already have a conflict. Could we redirect this conflict as a lemma in this case?

Reply:
Note this isn't specific to this PR (the other variants have this), but I agree we could reconsider this.  If anything, I believe we would just remove the guard and send another conflict.